### ISSUE TITLE
azimuth: bitbet-bolbel has out of date keys information for dozreg-taplud despite up to date +azimuth-block

### ISSUE THREAD

**Issue by zalberico** [urbit/urbit#6320]
- bitbet-bolbel and dozreg-taplud can't communicate
- this is because dozreg-taplud breached a couple of months ago and bitbet-bolbel does not have its new information
When running +keys on dozreg-toplud from bitbet-bolbel:
```
> +keys ~dozreg-toplud
[life=[~ 8] rift=[~ 6]]
%group-store: on-peek on
```

When running +keys on my ship and getting the correct data:
```
> +keys ~dozreg-toplud
[life=[~ 9] rift=[~ 7]]
```

+azimuth-block of dozreg-toplud: 16.614.040
+azimuth-block of bitbet-bolbel was at: 16.614.064

Something doesn't make sense?

I ran `azimuth-load` on bitbet-bolbel to see if it that will correct it but it'll take a bit to catch back up to latest azimuth state (currently it's on block 15.373.409 after running that).

**Comment by yapishu** [comment]
I also have this discrepancy from `~sitful-hatred`:

```
> +keys ~dozreg-toplud
[life=[~ 8] rift=[~ 6]]
```

**Comment by zalberico** [comment]
Fixed!

After running `-azimuth-load` on bitbet-bolbel and waiting a bit I now see:

```
> +keys ~dozreg-toplud
[life=[~ 9] rift=[~ 7]]
```

Still one thing doesn't make sense:
- Why was the life/rift state out of date when the block claimed to be up to date?

**Comment by zalberico** [comment]
@yapishu can you run `-azimuth-load` then it'll take ~30min to catch up.

**Comment by yapishu** [comment]
I ran `-azimuth-load` an hour ago but my azimuth seems stuck at 15.373.409 now :|

**Comment by zalberico** [comment]
It will usually catch up, but there are some other things we can do to debug.

`|install ~dister-norsyr-torryn %diagnose; to run it -diagnose!azimuth`

`-diagnose!azimuth`

You'll see output like:
```
last downloaded %azimuth block: #16.368.062 on ~2023.1.9..08.31.11
last processed %azimuth block: #16.133.035 on ~2022.12.7..13.10.47
installed %eth-watcher? %.y
running %eth-watcher? %.y
installed %azimuth? %.y
has %eth-watcher an %azimuth subscription? %.n
is %azimuth subscribed to %eth-watcher? %.y
...
(wait 5s. to run the thread again)
```

Can you summarize this and email into support@urbit.org with a reference to this issue? The UF can help you from there and that'll make sure it doesn't get dropped.

**Comment by yosoyubik** [comment]
> Why was the life/rift state out of date when the block claimed to be up to date?

@zalberico `+azimuth-block` prints the latest block seen by /app/eth-watcher, but that can be ahead of what /app/azimuth has seen (azimuth subscribes to eth-watcher so it receives the log updates some time after they are downloaded). Usually they are in sync, just a bit behind, unless it's the first time it receives logs (then it's way behind since you have downloaded a lot of recent logs) or there's another issue, possibly involving the apps' subscription (like this one: https://github.com/urbit/urbit/issues/5748#issuecomment-1374428270)
